source/uri-options/tests/srv-options.yml

```diff
@@ -22,10 +22,68 @@ tests:
       auth: ~
       options:
           srvMaxHosts: 2
+    - description: "SRV URI with negative integer for srvMaxHosts"
+      uri: "mongodb+srv://test1.test.build.10gen.cc/?srvMaxHosts=-1"
+      valid: false
+      warning: true
+      hosts: ~
+      auth: ~
+      options: {}
+    - description: "SRV URI with invalid type for srvMaxHosts"
+      uri: "mongodb+srv://test1.test.build.10gen.cc/?srvMaxHosts=foo"
+      valid: false
+      warning: true
+      hosts: ~
+      auth: ~
+      options: {}
     - description: "Non-SRV URI with srvMaxHosts"
       uri: "mongodb://example.com/?srvMaxHosts=2"
       valid: false
       warning: true
       hosts: ~
       auth: ~
       options: {}
+    # Note: Testing URI validation for srvMaxHosts conflicting with either
+    # loadBalanced=true or # replicaSet specified via TXT records is covered by
+    # the Initial DNS Seedlist Discovery test suite.
+    - description: "SRV URI with positive srvMaxHosts and replicaSet"
+      uri: "mongodb+srv://test1.test.build.10gen.cc/?srvMaxHosts=2&replicaSet=foo"
+      valid: false
+      warning: true
+      hosts: ~
+      auth: ~
+      options: {}
+    - description: "SRV URI with positive srvMaxHosts and loadBalanced=true"
+      uri: "mongodb+srv://test1.test.build.10gen.cc/?srvMaxHosts=2&loadBalanced=true"
+      valid: false
+      warning: true
+      hosts: ~
+      auth: ~
+      options: {}
+    - description: "SRV URI with positive srvMaxHosts and loadBalanced=false"
+      uri: "mongodb+srv://test1.test.build.10gen.cc/?srvMaxHosts=2&loadBalanced=false"
+      valid: true
+      warning: false
+      hosts: ~
+      auth: ~
+      options:
+          loadBalanced: false
+          srvMaxHosts: 2
+    - description: "SRV URI with srvMaxHosts=0 and replicaSet"
+      uri: "mongodb+srv://test1.test.build.10gen.cc/?srvMaxHosts=0&replicaSet=foo"
+      valid: true
+      warning: false
+      hosts: ~
+      auth: ~
+      options:
+          replicaSet: foo
+          srvMaxHosts: 0
+    - description: "SRV URI with srvMaxHosts=0 and loadBalanced=true"
+      uri: "mongodb+srv://test1.test.build.10gen.cc/?srvMaxHosts=0&loadBalanced=true"
```

Comment:
Ah, I wasn't aware that the URI Options tests were actually doing SRV lookups. I suppose this isn't really addressed by the test README since `srv-options.yml` is the first file to actually use `mongodb+srv://`.